File: engine/engine-model/src/main/scala/tools/packet/guild/NewGuildMember.scala
package tools.packet.guild

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class NewGuildMember(private var _guildId: Int, private var _characterId: Int, private var _name: String,
                     private var _jobId: Int, private var _level: Int, private var _guildRank: Int,
                     private var _online: Boolean) extends PacketInput {
  def guildId: Int = _guildId

  def characterId: Int = _characterId

  def name: String = _name

  def jobId: Int = _jobId

  def level: Int = _level

  def guildRank: Int = _guildRank

  def online: Boolean = _online

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/NewGuildMember.java
package tools.packet.guild;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record NewGuildMember(Integer guildId, Integer characterId, String name, Integer jobId, Integer level,
                             Integer guildRank, Boolean online) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}